Extension/src/LanguageServer/ui.ts

```diff
@@ -344,6 +344,34 @@ export class UI {
         return (selection) ? selection.index : -1;
     }
 
+    public async showSelectCompiler(paths: string[]): Promise<number> {
+        const options: vscode.QuickPickOptions = {};
+        options.placeHolder = localize("select.compile.commands", "Select a compiler to configure for IntelliSense");
+
+        const items: IndexableQuickPickItem[] = [];
+        for (let i: number = 0; i < paths.length; i++) {
+            let compiler: string | undefined;
+            let isCompiler: boolean = false;
+            if (paths[i].indexOf("\\") > 0) {
+                if (paths[i].split("\\").pop() !== undefined) {
+                    compiler = paths[i].split("\\").pop();
+                    isCompiler = true;
```

Comment:
`isCompiler` needed or would it be sufficient to check if `compiler` was set?

Reply:
This has to do with the way that I am adding extra options to the quick pick. There will non compiler values in the quick pick that allow users to do other things. The only way to make these visible in the quick pick is to make them a label so I have to distinguish whether the compiler variable is a compiler. Perhaps I will change the variable from "compiler" to "option"